Feature/init

ctrlaltd3v opened the pull request.

ctrlaltd3v committed:
.

ctrlaltd3v committed:
initial version

erlendoksvoll commented on src/Altinn.Dan.Plugin.Statensvegvesen/Clients/Svv.cs:
```diff
@@ -0,0 +1,69 @@
+                var technicalApproval = kjoretoyResponse.Kjoretoy?.Godkjenning?.TekniskGodkjenning;
+                internalResponse.Vehicles.Add(new Vehicle
+                {
+                    RegNr = kjoretoyResponse.Kjoretoy?.KjoretoyId?.Kjennemerke,
```
RegNo is the English abbreviation

erlendoksvoll commented on src/Altinn.Dan.Plugin.Statensvegvesen/Metadata.cs:
```diff
@@ -0,0 +1,48 @@
+                    {
+                        EvidenceValueName = "default",
+                        ValueType = EvidenceValueType.JsonSchema,
+                        JsonSchemaDefintion = JsonConvert.SerializeObject(new SvvResponse())
```
Maybe something like JsonSchema.FromType<Type>().ToJson(Newtonsoft.Json.Formatting.Indented)

ctrlaltd3v replied:
Absolutely, it should be a SCHEMA.
Fix commited.

ctrlaltd3v committed:
Changes after PR.
JsonSchemaDefintion(JsonSchemaDefinition) is now a JSON Schema.
Corrected handling of missing dates from external API.

ctrlaltd3v commented:
Take a look at the changes for handling of missing dates.

erlendoksvoll commented on src/Altinn.Dan.Plugin.Statensvegvesen/Utils/Utils.cs:
```diff
@@ -0,0 +1,74 @@
+        if (externalSource == "svv")
+            switch (ssn)
+            {
+                case "12113701569":
```
Could just remove the switch and keep only the svv values? Perhaps I should do the same for banking ... :|

ctrlaltd3v replied:
Agree, looking into it :-)

ctrlaltd3v replied:
Removed all of this :-D

ctrlaltd3v committed:
Using NSwag only to generate models, homebrewed client

ctrlaltd3v commented:
How about this contraption? Homemade client, I guess it needs a bit of honing, but it's a lot smaller than the generated one. (Maybe easier to maintain..)

erlendoksvoll commented on src/Altinn.Dan.Plugin.Statensvegvesen/Clients/SvvClient.cs:
```diff
@@ -0,0 +1,66 @@
+                    case HttpStatusCode.NoContent:
+                    {
+                        throw new EvidenceSourcePermanentClientException(Metadata.ERROR_NO_VEHICLES_FOUND, $"No vehicles found");
+                    }
```
The api returns nocontent when no vehicles are found?

ctrlaltd3v replied:
Yup.
From the spec:
"responses":
          "200": 
            "description": "Søk gjennomført, responsen inneholder eventuelle treff og/eller responskoder"
          "204": 
            "description": "Ingen kjøretøy funnet"
          "400": 
            "description": "Parameterfeil, responsen inneholder kun beskrivelse av feil som responskoder"

erlendoksvoll replied:
Maybe we still should return an empty list instead of a permanentexception?

ctrlaltd3v committed:
Added some tests, small adjustments

ctrlaltd3v committed:
More changes after PR, added test, updated NuGets

ctrlaltd3v committed:
Removed commented out code
